SwiftCharts/Axis/ChartAxisLayer.swift

```diff
@@ -8,20 +8,48 @@
 
 import UIKit
 
+/// A protocol for all axis layers
 public protocol ChartAxisLayer: ChartLayer {
-        
+
+    /// The location of the smallest axis value
     var p1: CGPoint {get}
+
+    /// The location of the largest axis value
     var p2: CGPoint {get}
+
+    /// The axis values shown in the layer
     var axisValues: [ChartAxisValue] {get}
+
+    /// The bounding rectangle of the layer
```

Comment:
This is more exactly the bounding rectangle of the axis with it's labels, including spacing between the axis and labels, multiple labels or rotation (in case of x axis). This is for example how the rect's height of x-axis is calculated: https://github.com/i-schuetz/SwiftCharts/blob/546a5deb92d773edfb110d9319e89acc61bf7a70/SwiftCharts/Axis/ChartAxisXLayerDefault.swift#L28